model/task/task.go

```diff
@@ -2750,76 +2474,102 @@ func (t *Task) makeArchivedTask() *Task {
 
 // Aggregation
 
-// PopulateTestResults returns the task with both old (embedded in the tasks
-// collection) and new (from the testresults collection) OR Cedar test results
-// merged into the Task's LocalTestResults field.
+// PopulateTestResults populates the task's LocalTestResults field with any
+// test results the task may have. If the results are already populated, this
+// function no-ops.
 func (t *Task) PopulateTestResults() error {
```

Comment:
Yeah its for caching, I think the some of the trigger functionality may need to cache them. We can revisit removing this function entirely after PR. Since there are a significant amount of changes already, I think its best to play it safe and maintain this functionality as is.